### ISSUE TITLE
'yamlParser is not defined' error on latest docker image

### ISSUE THREAD

**Issue by eedwardsdisco** [projectatomic/dockerfile_lint#75]
According to docs it should be optional.

```
projectatomic/dockerfile-lint@sha256:379fba187c13e6f5a14e97405bb1f0d6b7fdfe5f806d876ce2240c1eda45a104
```

```
running dockerfile_lint -f dockerfile-dir/Dockerfile
/opt/dockerfile_lint/bin/dockerfile_lint:59
    console.log(yamlParser.dump(rules));
                ^

ReferenceError: yamlParser is not defined
    at Object.<anonymous> (/opt/dockerfile_lint/bin/dockerfile_lint:59:17)
    at Module._compile (module.js:570:32)
    at Object.Module._extensions..js (module.js:579:10)
    at Module.load (module.js:487:32)
    at tryModuleLoad (module.js:446:12)
    at Function.Module._load (module.js:438:3)
    at Module.runMain (module.js:604:10)
    at run (bootstrap_node.js:394:7)
    at startup (bootstrap_node.js:149:9)
    at bootstrap_node.js:509:3
```

**Comment by eedwardsdisco** [comment]
Actually looking at the error it's not even finding `yamlParser`.

This is straight off the image so?

**Comment by eedwardsdisco** [comment]
Yeah rules was a red herring, specifying explicit rules does nothing.

**Comment by lphiri** [comment]
hi @eedwardsdisco  - not quite sure what the issue is. Can you explicitly  outline the steps to reproduce the problem?

**Comment by eedwardsdisco** [comment]
@lphiri 

Sure thing. I'm running concourse.ci inside docker-compose which mounts a docker-image using garden to run tasks.

my task definition looks like this

```
---
platform: linux

image_resource:
  type: docker-image
  source:
    repository: projectatomic/dockerfile-lint

inputs:
  - name: dockerfile-dir

run:
  path: dockerfile_lint
  args:
    - -f dockerfile-dir/Dockerfile
```

and I can execute it like this

```
$ fly execute -c ci/concourse/tasks/unit.yml -t main --input=dockerfile-dir=src/main/docker/buildrunner
```

So, based on your `Dockerfile`, I land in the `/root` directory.

I provide the directory for my Dockerfile, and concourse uploads the contents into `/root/dockerfile-dir` for me when I run the execute.

Due to the `Dockerfile`, I have `dockerfile_init` available in the `$PATH`.

Thus I can execute `dockerfile_lint` from my working directory of `/root` specifying `-f dockerfile-dir/Dockerfile` which points to the `Dockerfile` that was uploaded by concourse that I'm trying to lint.

**Comment by eedwardsdisco** [comment]
OK, I solved it.

I'll post my solution for posterity.

Appears to be due to how dockerfile_lint or node handles arguments.

Normal BASH style arguments expect both the flag and the value to be in the same 'argument', e.g.:

`./myscript "-h arg_value_for_h" "-a arg_value_for_a"`

however, it appears that for whatever reason I have to specify them as separate arguments when calling it from concourse, so more like this:

`./myscript "-h" "arg_value_for_h" "-a" "arg_value_for_a"`

where each argument is a 'separate' argument.

I verified this in my shell on my mac, as well (real path redacted):

```
$ ./dockerfile_lint "-f ../../valid/path/to/Dockerfile"
ERROR: Dockerfile not found -> Dockerfile
```
